src/indexer/segment_updater.rs

```diff
@@ -82,6 +84,7 @@ fn save_metas(metas: &IndexMeta, directory: &dyn Directory) -> crate::Result<()>
         io::ErrorKind::Other,
         msg.unwrap_or_else(|| "Undefined".to_string())
     ))));
+    directory.sync_directory()?;
```

Comment:
sync before the atomic write?

Reply:
Yes. Again this is a directory sync. Its point is to make sure all of the component files we created  are durably created.

Reply:
I thought that the atomic write may cause a dangling file if it only calls sync_data, but it doesn't because in the mmap_dir we call sync_directory, so we can remove that call before?

```
    fn atomic_write(&self, path: &Path, content: &[u8]) -> io::Result<()> {
        debug!("Atomic Write {:?}", path);
        let full_path = self.resolve_path(path);
        atomic_write(&full_path, content)?;
        self.sync_directory()
    }
```

Reply:
Yes! So the idea is we just need to be sure the files are durably persisted right before the commit, so we flush the directory right before the commit atomic_write

Reply:
I think we could save some calls in sync_directory after atomic_write. We need to sync the file only once.

This may require some bookkeeping. `atomic_write` seems to be used only by managed and meta.json, so we could think about making a specialized api

Reply:
I am not sure what you are suggesting exactly. There are many ways to reduce the number of fsync. 

Probably the simplest, least leaking solutions are:
- add a `open_writes(...)` method that takes a list of path instead of a single one.
- remove the managed.json file and rely on listing the directory.
 
A more involved solution but drastically more efficient could be to transparently bundle files into one if they share a prefix and they are small.

With that we would have only
- 3 file flush
- 1 directory flush

If we remove the managed.json too, we could get down to 2 file flush and 1 directory flush.
The cost would be of < 3ms per commit on an SSD.

That might be enough for NRT use cases (usually 50ms or so) without doing all of the NRT plumbing.

Reply:
I'm not sure how multiple open_writes would reduce the amount of fsync calls.

`open_write` returns a SafeFileWriter which calls sync_all on terminate.
In both cases we have one sync_data call per file on commit?